webapp/Connector/src/model/ChartData.js

```diff
@@ -457,7 +461,12 @@ Ext.define('Connector.model.ChartData', {
                 xname: xa.label,
                 yname: ya.label,
                 colorname: ca.label,
-                rowKey: rowKey
+                rowKey: rowKey,
+                querySrc: ya.query,
```

Comment:
This is too much hardcoding. 
- Since allowForGrouping is already defined at measure level, it should iterate through measures to find the fields needed for grouping, instead of hardcode here again for PKMAb. This allows other assay to define their own line group as well. 
- a single [groupField] can be built and passed to plot, instead of having plot rebuild the group value.

Reply:
Great suggestion, now done.